merkle/index.go

```diff
@@ -2,52 +2,83 @@ package merkle
 
 // siblingIndecies returns indecies of sibling elements
 func siblingIndecies(idxs []uint64) []uint64 {
+
 	indicesCount := len(idxs)
 	siblings := make([]uint64, indicesCount)
+
+	// append all sibling indices to result
 	for i := 0; i < indicesCount; i++ {
 		siblings[i] = siblingIndex(idxs[i])
 	}
+
 	return siblings
 }
 
 // parentIndecies returns indecies of parent elements
 func parentIndecies(idxs []uint64) []uint64 {
```

Comment:
can you please change `idx` to a more meaningful name? maybe something like `levelIndices`?